modules/issue_tracker/php/uploadhelper.class.inc

```diff
@@ -0,0 +1,254 @@
+            'issueID'     => $this->_values['issueID'],
+            'file_hash'   => $fileToUpload->file_hash,
+            'date_added'  => date('Y-m-d h:i:s', time()),
+            'file_name'   => $fileToUpload->file,
+            'deleted'     => 0,
+            'user'        => $fileToUpload->inserted_by,
+            'description' => $fileToUpload->description,
+            'file_size'   => $fileToUpload->size,
+            'mime_type'   => $fileToUpload->mime_type
+        );
+        try {
+            $DB->insert('issues_attachments', $values);
+            // Move file from tmp to issue_tracker attachment directory.
+            $path = $fileToUpload->full_path . '/' . $fileToUpload->file_hash;
+            move_uploaded_file($_FILES['file']['tmp_name'], $path);
+        } catch (\DatabaseException $e) {
+            $this->endWithFailure();
+            error_log("DatabaseException: $e");
+            $this->_response = json_encode(
+                array(
+                    'message' => 'File registration failed.
+                    Is the description empty?',
+                    'error'   => true,
+                )
+            );
+        }
+    }
+
+    /**
+     * Get mime_type details for the attachment file.
+     *
+     * @param string $file the file name with extension.
+     *
+     * @return string $mime_type
+     */
+    function getFileMimeType(string $file) : string
+    {
+        $known_mime_types = array(
+            'htm'  => 'text/html',
+            'csv'  => 'text/csv',
+            'exe'  => 'application/octet-stream',
+            'zip'  => 'application/zip',
+            'doc'  => 'application/msword',
+            'jpg'  => 'image/jpg',
+            'php'  => 'text/plain',
+            'xls'  => 'application/vnd.ms-excel',
+            'ppt'  => 'application/vnd.ms-powerpoint',
+            'gif'  => 'image/gif',
+            'pdf'  => 'application/pdf',
+            'txt'  => 'text/plain',
+            'html' => 'text/html',
+            'png'  => 'image/png',
+            'jpeg' => 'image/jpg',
+        );
+        if (strpos($file, '.') !== false) {
+            $file_extension = strtolower(substr(strrchr($file, '.'), 1));
+            if (array_key_exists($file_extension, $known_mime_types)) {
+                $mime_type = $known_mime_types[$file_extension];
+            } else {
+                $mime_type = 'application/force-download';
+            }
+        } else {
+            $mime_type = 'application/force-download';
+        }
+        return $mime_type;
+    }
+
+    /**
+     * Create a sha256 string from file_content.
+     *
+     * @param string $filePath the path to the file.
+     *
+     * @return string
+     */
+    function generateHash(string $filePath) : string
+    {
+        return hash_file('sha256', $filePath);
+    }
+
+    /**
+     * Begins the transaction for the file attachment upload.
+     *
+     * @return void
+     */
+    function begin() : void
```

Comment:
I don't think these custom DB functions are necessary.

Reply:
@johnsaigle the begin, endWithFailure, endWithSuccess, are being used to handle if there is an error with the file being saved to the server. I believe it's similar to an atomic insert for the db. In case something happens with the server, the db won't have the file info inserted. So I believe it's necessary.

Reply:
Why not just call the Database functions directly?

Reply:
@johnsaigle I'm not sure what you mean.

Reply:
Instead of creating a `begin()` function here why not just call the Database::beginTransaction function directly on line 88?